src/algorithms/reco/ChargedParticleSelector.cc

```diff
@@ -3,6 +3,8 @@
 
 #pragma once
 
+#include <extensions/jana/JChainMultifactoryT.h>
```

Comment:
I can see the benefit of uniformity for a future where all algorithms behave the same way, but in that future we hopefully have concepts or other SFINAE magic (https://en.cppreference.com/w/cpp/types/is_base_of) to determine if we actually inherit from `WithPodConfig.h`

Reply:
For uniformity? I don't know. Feel free to commit what's needed.

Reply:
Oh, nvm. I understand, it makes no sense to have this include here.

Reply:
This would be the first such dependence, though.

https://eicrecon.epic-eic.org/pr/928/doxygen/dir_e6df591b0639d4c4807ef64d058833e2.html
![image](https://github.com/eic/EICrecon/assets/4656391/09260650-fd08-4d9b-ab54-378eb44b6bb3)

with only the following links (in current main):
https://eicrecon.epic-eic.org/pr/928/doxygen/dir_000075_000032.html

File in src/algorithms/tracking	Includes file in src/extensions
[ActsGeometryProvider.cc](https://eicrecon.epic-eic.org/pr/928/doxygen/ActsGeometryProvider_8cc.html)	[spdlog](https://eicrecon.epic-eic.org/pr/928/doxygen/dir_e7436a078012916a06198d03fe3458c4.html) / [SpdlogFormatters.h](https://eicrecon.epic-eic.org/pr/928/doxygen/SpdlogFormatters_8h.html)
[ActsGeometryProvider.cc](https://eicrecon.epic-eic.org/pr/928/doxygen/ActsGeometryProvider_8cc.html)	[spdlog](https://eicrecon.epic-eic.org/pr/928/doxygen/dir_e7436a078012916a06198d03fe3458c4.html) / [SpdlogToActs.h](https://eicrecon.epic-eic.org/pr/928/doxygen/SpdlogToActs_8h.html)
[CKFTracking.cc](https://eicrecon.epic-eic.org/pr/928/doxygen/CKFTracking_8cc.html)	[spdlog](https://eicrecon.epic-eic.org/pr/928/doxygen/dir_e7436a078012916a06198d03fe3458c4.html) / [SpdlogFormatters.h](https://eicrecon.epic-eic.org/pr/928/doxygen/SpdlogFormatters_8h.html)
[CKFTracking.cc](https://eicrecon.epic-eic.org/pr/928/doxygen/CKFTracking_8cc.html)	[spdlog](https://eicrecon.epic-eic.org/pr/928/doxygen/dir_e7436a078012916a06198d03fe3458c4.html) / [SpdlogToActs.h](https://eicrecon.epic-eic.org/pr/928/doxygen/SpdlogToActs_8h.html)
[IterativeVertexFinder.cc](https://eicrecon.epic-eic.org/pr/928/doxygen/IterativeVertexFinder_8cc.html)	[spdlog](https://eicrecon.epic-eic.org/pr/928/doxygen/dir_e7436a078012916a06198d03fe3458c4.html) / [SpdlogFormatters.h](https://eicrecon.epic-eic.org/pr/928/doxygen/SpdlogFormatters_8h.html)
[IterativeVertexFinder.cc](https://eicrecon.epic-eic.org/pr/928/doxygen/IterativeVertexFinder_8cc.html)	[spdlog](https://eicrecon.epic-eic.org/pr/928/doxygen/dir_e7436a078012916a06198d03fe3458c4.html) / [SpdlogToActs.h](https://eicrecon.epic-eic.org/pr/928/doxygen/SpdlogToActs_8h.html)
[TrackParamTruthInit.cc](https://eicrecon.epic-eic.org/pr/928/doxygen/TrackParamTruthInit_8cc.html)	[spdlog](https://eicrecon.epic-eic.org/pr/928/doxygen/dir_e7436a078012916a06198d03fe3458c4.html) / [SpdlogFormatters.h](https://eicrecon.epic-eic.org/pr/928/doxygen/SpdlogFormatters_8h.html)
[TrackProjector.cc](https://eicrecon.epic-eic.org/pr/928/doxygen/TrackProjector_8cc.html)	[spdlog](https://eicrecon.epic-eic.org/pr/928/doxygen/dir_e7436a078012916a06198d03fe3458c4.html) / [SpdlogFormatters.h](https://eicrecon.epic-eic.org/pr/928/doxygen/SpdlogFormatters_8h.html)

Reply:
After 5bed9a8 there should be no more need for this.

```suggestion
```

Reply:
> This would be the first such dependence, though.

Not true. Some header-only algorithms like TrackerHitCollector.h must be pulling NoConfig from JANA.

Reply:
No, the `NoConfig` is just added at the factory stage, e.g. https://github.com/eic/EICrecon/blob/main/src/factories/tracking/TrackerHitCollector_factory.h#L17. That (no)config is never applied to the algorithm. That's what I meant with uniformity. If JOmniFactoryT always applies the config to the algorithm, then we need to make sure that each algorithm can accept it, and they all must expose the `applyConfig` in `WithPodConfig` (or we need to be clever and SFINAE it out).